oap-server/server-receiver-plugin/skywalking-sharing-server-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/sharing/server/ReceiverGRPCHandlerRegister.java

```diff
@@ -18,22 +18,39 @@
 
 package org.apache.skywalking.oap.server.receiver.sharing.server;
 
-import io.grpc.*;
+import io.grpc.BindableService;
+import io.grpc.ServerInterceptor;
+import io.grpc.ServerInterceptors;
+import io.grpc.ServerServiceDefinition;
+import java.util.Objects;
 import lombok.Setter;
 import org.apache.skywalking.oap.server.core.server.GRPCHandlerRegister;
 
 /**
- * @author peng-yongsheng
+ * @author peng-yongsheng, jian.tan
  */
 public class ReceiverGRPCHandlerRegister implements GRPCHandlerRegister {
 
     @Setter private GRPCHandlerRegister grpcHandlerRegister;
+    private ServerInterceptor interceptor;
 
     @Override public void addHandler(BindableService handler) {
-        grpcHandlerRegister.addHandler(handler);
+        if (Objects.isNull(interceptor)) {
+            grpcHandlerRegister.addHandler(handler);
+        } else {
+            grpcHandlerRegister.addHandler(handlerInterceptorBind(handler, interceptor));
+        }
```

Comment:
So you have an implicit prerequisite here: the `addFilter` method must be called before adding all handlers (`addHandler`), otherwise the filter doesn't fully take effect, right? If so, this prerequisite must be pointed out in the `interface`, but I don't think this is a good design, I'd suggest moving the interceptor to:

https://github.com/apache/skywalking/blob/6f90545a537d3830d0fcc1b82f51b444f848b5d4/oap-server/server-library/library-server/src/main/java/org/apache/skywalking/oap/server/library/server/grpc/GRPCServer.java#L109

what do you think @wu-sheng

Reply:
Token authentication filter should only be activated in sharding server module. Moving to `GRPCServer` may not very suitable, in my thought.

Reply:
> Token authentication filter should only be activated in sharding server module. Moving to `GRPCServer` may not very suitable, in my thought.

Well I missed that